### PR TITLE
PRL-3771: Upload supporting documents screen

### PR DESCRIPTION
**Pull request by hcarslaw**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the keep-helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/app/case/case.ts [hunk 1]
```diff
@@ -598,6 +598,7 @@ export interface Case {
   awp_reasonCantBeInformed?: string;
   awp_uploadedApplicationForms?: DocumentInfo[];
   awp_hasSupportingDocuments?: YesOrNo;
+  awp_uploadedSupportingDocuments?: DocumentInfo[];
```
Comment (vivek-sekhar) [inline src/main/app/case/case.ts:601]: just use supportingDocuments

File: src/main/steps/application-within-proceedings/document-upload/postController.ts [hunk 2]
```diff
@@ -23,29 +23,17 @@ export default class UploadDocumentController extends PostController<AnyObject>
...
     const { applicationType, applicationReason } = req.params;
 
-    if (
-      req.body.onlyContinue &&
-      req.session.userCase.awp_uploadedApplicationForms &&
-      req.session.userCase.awp_uploadedApplicationForms.length > 0
-    ) {
+    const isSupportingDocuments = req.route.path === APPLICATION_WITHIN_PROCEEDINGS_SUPPORTING_DOCUMENT_UPLOAD;
+    const areFilesUploaded = isSupportingDocuments
+      ? req.session.userCase.awp_uploadedSupportingDocuments &&
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/document-upload/postController.ts:28]: you can do this in 1 condition with optional property checking

File: src/main/steps/application-within-proceedings/document-upload/routeGuard.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ export const routeGuard = {
           await caseApi(userDetails, req.locals.logger).deleteDocument(removeId.toString());
         } catch (error) {
           res.json(error);
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/document-upload/routeGuard.ts:26]: why res.json required?

File: src/main/steps/application-within-proceedings/inform-other-parties/content.ts [hunk 4]
```diff
@@ -85,7 +87,7 @@ export const form: FormContent = {
   },
   link: {
     classes: 'govuk-!-margin-left-3',
-    href: '/application-within-proceedings/list-of-applications/1',
+    href: applyParms(APPLICATION_WITHIN_PROCEEDINGS_LIST_OF_APPLICATIONS, { pageNumber: '1' }),
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/inform-other-parties/content.ts:90]: please put this in a util and call it in all content.ts

File: src/main/steps/application-within-proceedings/supporting-document-upload/routeGuard.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        );
+      }
+
+      if (documentToDelete) {
+        try {
+          req.session.errors = [];
+          const userDetails = req?.session?.user;
+          await caseApi(userDetails, req.locals.logger).deleteDocument(removeId.toString());
+        } catch (error) {
+          res.json(error);
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/supporting-document-upload/routeGuard.ts:26]: why res.json required?

Comment (hcarslaw) [reply src/main/steps/application-within-proceedings/supporting-document-upload/routeGuard.ts]: It's not needed so I've removed it

File: src/main/steps/application-within-proceedings/document-upload/routeGuard.ts (hunk 3 above)
Comment (hcarslaw) [reply src/main/steps/application-within-proceedings/document-upload/routeGuard.ts]: It's not needed so I've removed it

File: src/main/app/case/case.ts (hunk 1 above)
Comment (hcarslaw) [reply src/main/app/case/case.ts]: Changed to supportingDocuments

File: src/main/steps/application-within-proceedings/document-upload/postController.ts (hunk 2 above)
Comment (hcarslaw) [reply src/main/steps/application-within-proceedings/document-upload/postController.ts]: Using one condition for this now

File: src/main/steps/application-within-proceedings/inform-other-parties/content.ts (hunk 4 above)
Comment (hcarslaw) [reply src/main/steps/application-within-proceedings/inform-other-parties/content.ts]: Changed to use constant in utils

File: src/main/app/case/case.ts [hunk 6]
```diff
@@ -598,7 +598,7 @@ export interface Case {
   awp_reasonCantBeInformed?: string;
   awp_uploadedApplicationForms?: DocumentInfo[];
   awp_hasSupportingDocuments?: YesOrNo;
-  awp_uploadedSupportingDocuments?: DocumentInfo[];
+  supportingDocuments?: DocumentInfo[];
```
Comment (vivek-sekhar) [inline src/main/app/case/case.ts:603]: prefix awp_

File: src/main/steps/application-within-proceedings/utils.ts [hunk 7]
```diff
@@ -409,3 +411,7 @@ export const isValidApplicationReason = (
 
   return false;
 };
+
+export const AWP_APPLICATION_LIST_FIRST_PAGE = applyParms(APPLICATION_WITHIN_PROCEEDINGS_LIST_OF_APPLICATIONS, {
```
Comment (vivek-sekhar) [inline src/main/steps/application-within-proceedings/utils.ts:415]: Just change the const name to APPLICATION_SIGNPOSTING_URL

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

**Pull request merged**